### DISCUSSION TITLE
Any plans for AWS Bedrock ?

### DISCUSSION THREAD

**Discussion by maximelebastard** [janhq/jan#3845] [Feature Ideas]
Hello there,

AWS Bedrock is an Amazon service providing inference for models like Llama, Cohere, Claude, Titan etc...
For AWS users, it is pretty convienient as it is priced per token (and pretty cheap), included in AWS billing, with no data retention (for pro usages).

Is there any obstacle to integrate it ? Auth is usually done through an AWS Access Key

If I had to write a PR for it, where should I look at first ? (is there a contributing guide to adding a model provider or an example PR)

Best

**Comment by dan-menlo** [comment]
@maximelebastard We are currently refactoring our "Provider" API in Sprint 23: 

- See https://github.com/janhq/jan/issues/3824
- We will be publishing a clear "Provider Extension API", that allows the community to build your own Provider extensions
- AWS Bedrock is definitely one that we need to support!

cc @gabrielle-ong who was formerly from AWS 😂
